File: nodejs/LakeView.js
// Example For Get Lake View API by using request in nodejs
// written by yeony kim

process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";

const express = require('express');
const app = express();
const API_KEY = "YOUR_API_TOKEN";
const LAKE_ID = "YOUR_LAKE_ID"
const URL = "https://api.machlake.com/lakes/" + LAKE_ID

// CASE - GET LAKE view

app.get(
    '/get_lake_view', function(req, res) {
        var request = require('request');
        var option = {
            url: URL,
            headers: {
                'Content-Type': 'application/json',
                'x-api-key': API_KEY
            },
        };
        request.get( option, function(error, response, body) {
            if (!error) {
                res.writeHead(200);
                res.end(body);
                console.log(body) // {"data":{"access_control_list":[{"cidr":"0.0.0.0/0","name":"default"}],"lake_id":"YOUR_LAKE_ID","lake_info":{"create_date":"2021-10-14 01:20:34","lake_name":"sample_lake","lake_plan":"basic","lake_type":"Normal","region":"ap-northeast-2","timezone":"Asia/Seoul","update_date":"0"},"lake_status":{"count_of_tag":0,"count_of_value":0,"query_call":0,"state":"running","storage":7456575488,"traffic":1990097},"plan_limit":{"default_tag_count":10000,"limit_append_tag":1000,"limit_append_value":100,"limit_select_tag":1000,"limit_select_value":100,"max_concurrent":100000,"max_network":10737418240,"max_query":10000,"max_storage":21474836480,"max_tag_count":100000},"tag_schema":[{"col_length":40,"col_name":"name","col_type":"varchar"}],"value_schema":[{"col_length":0,"col_name":"time","col_type":"datetime"},{"col_length":0,"col_name":"value","col_type":"double"}]},"status":"success"}
            } else {
                console.log(error)
            }
        })
    }
)

app.listen(8888, function() {
    console.log('http://127.0.0.1:8888/get_lake_view is result for get LAKE view')
})

File: python/LakeView.py
"""
Example For Get Lake View API by using request in python
written by yeony kim
"""

import requests

API_KEY = "YOUR_API_TOKEN"
LAKE_ID = "YOUR_LAKE_ID"

URL = f"https://api.machlake.com/v1/lakes/{LAKE_ID}"

headers = {
    "Content-Type": "application/json",
    "x-api-key": API_KEY
}


# ------------------------------------------------------------------------------------------------- #

# CASE - GET LAKE List

response = requests.get(URL, headers=headers, verify=False)
print(response.content.decode("utf-8"))

# Return Format
# {
#     "success":true,
#     "reason":"get lake info success",
#     "data":{
#         "lake_id":"xbcd0001",
#         "lake_info":{
#             "lake_name":"sample_lake",
#             "lake_plan":"basic",
#             "lake_type":"N",
#             "region":"aws1.kor",
#             "timezone":"America/Los_Angeles",
#             "create_time":"2021-09-30 10:30:05",
#             "update_time":"2021-10-01 14:59:12"
#         },
#         "lake_status":{
#             "count_of_tag":128,
#             "count_of_value":123456,
#             "query_call":0,
#             "traffic":4743829,
#             "storage":3032870912,
#             "state":"running" 
#         },
#         "plan_limit":{
#             "max_tag":1000,
#             "max_query":10000,
#             "max_disk":32212254720,
#             "max_concurrent":100000,
#             "limit_select_tag":1000,
#             "limit_select_value":100,
#             "limit_append_value":100,
#             "limit_append_tag":1000,
#             "default_tag_count":10000
#         },
#         "tag_schema":[
#             {
#                 "col_name":"name",
#                 "col_type":"varchar",
#                 "col_length":40
#             }
#         ],
#         "value_schema":[
#             {
#                 "col_name":"time",
#                 "col_type":"datetime"
#             },
#             {
#                 "col_name":"value",
#                 "col_type":"double"
#             }
#         ]
#     }
# }
